net/tsaddr/tsaddr.go

```diff
@@ -138,3 +138,50 @@ type onceIP struct {
 	sync.Once
 	v netaddr.IP
 }
+
+// NewContainsIPFunc returns a func that reports whether ip is in addrs.
+//
+// It's optimized for the cases of addrs being empty and addrs
+// containing 1 or 2 single-IP prefixes (such as one IPv4 address and
```

Comment:
Looks like it's fast for any number of single-IP addresses. It's ranges that are slow.

Reply:
well, _really_ fast for 1 or 2. the map lookup is much slower.